File: 2661. First Completely Painted Row or Column/Solution.cpp
class Solution {
public:
    int firstCompleteIndex(vector<int>& arr, vector<vector<int>>& mat) {
        int m = mat.size(), n = mat[0].size(), low = m * n;
        vector<int> cols(n, 0);
        unordered_map<int, int> time;
        for (int i = 0; i < m * n; i++) time[arr[i]] = i;
        for (int i = 0; i < m; i++) {
            int rows = 0;
            for (int j = 0; j < n; j++) {
                rows = max(rows, time[mat[i][j]]);
                cols[j] = max(cols[j], time[mat[i][j]]);
            }
            low = min(low, rows);
        }
        for (int c : cols) low = min(low, c);
        return low;
    }
};

File: 2661. First Completely Painted Row or Column/Solution.py
class Solution:
    def firstCompleteIndex(self, arr: List[int], mat: List[List[int]]) -> int:
        m, n, low = len(mat), len(mat[0]), len(mat) * len(mat[0])
        cols = [0 for i in range(n)]
        time = {arr[i] : i for i in range(m * n)}
        for i in range(m):
            rows = 0
            for j in range(n):
                rows = max(rows, time[mat[i][j]])
                cols[j] = max(cols[j], time[mat[i][j]])
            low = min(low, rows)
        for c in cols: low = min(low, c)
        return low